### PR TITLE
[16.0][MIG] delivery_state: Migration to 16.0

### PR DESCRIPTION
**Pull request by chienandalu**
- Standard migration
- Delivery state term correction
- Implement fix in https://github.com/OCA/delivery-carrier/pull/555

cc @Tecnativa TT40419

please review @victoralmau @pedrobaeza

### CODE REVIEW

**Review by victoralmau — APPROVED** [review]
Code review OK.

File: delivery_state/models/stock_picking.py [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+    )
+    tracking_state_history = fields.Text(
+        readonly=True,
+    )
+    delivery_state = fields.Selection(
+        selection=[
+            ("shipping_recorded_in_carrier", "Shipping recorded in carrier"),
+            ("in_transit", "In transit"),
+            ("canceled_shipment", "Canceled shipment"),
+            ("incidence", "Incidence"),
```
Comment (rousseldenis) [inline delivery_state/models/stock_picking.py]: I think you wanted to say 'Incident' instead

Comment (chienandalu) [reply delivery_state/models/stock_picking.py]: Oops you're right. An spanish false friend.

Comment (chienandalu) [reply delivery_state/models/stock_picking.py]: Fixed!

File: delivery_state/i18n/es.po [hunk 2]
```diff
@@ -120,8 +120,8 @@ msgid "In transit"
 msgstr "En tránsito"
 
 #. module: delivery_state
-#: model:ir.model.fields.selection,name:delivery_state.selection__stock_picking__delivery_state__incidence
-msgid "Incidence"
+#: model:ir.model.fields.selection,name:delivery_state.selection__stock_picking__delivery_state__incident
+msgid "incident"
```
Comment (pedrobaeza) [inline delivery_state/i18n/es.po]:
```suggestion
msgid "Incident"
```

File: delivery_state/i18n/delivery_state.pot [hunk 3]
```diff
@@ -101,8 +101,8 @@ msgid "In transit"
 msgstr ""
 
 #. module: delivery_state
-#: model:ir.model.fields.selection,name:delivery_state.selection__stock_picking__delivery_state__incidence
-msgid "Incidence"
+#: model:ir.model.fields.selection,name:delivery_state.selection__stock_picking__delivery_state__incident
+msgid "incident"
```
Comment (pedrobaeza) [inline delivery_state/i18n/delivery_state.pot]: Beware of manual changes:

```suggestion
msgid "Incident"
```

File: delivery_state/models/stock_picking.py [hunk 4]
```diff
@@ -66,14 +66,7 @@ def _update_delivery_state(self):
                 ("delivery_type", "not in", [False, "fixed", "base_one_rule"]),
             ]
         )
-        delivery_types = self.mapped("delivery_type")
-        # Split them by delivery type so we can ignore those without the
```
Comment (pedrobaeza) [inline delivery_state/models/stock_picking.py]: Why this is not needed anymore?

Comment (pedrobaeza) [reply delivery_state/models/stock_picking.py]: Answered with #555

File: delivery_state/i18n/delivery_state.pot (hunk 3 above)
Comment (chienandalu) [reply delivery_state/i18n/delivery_state.pot]: Thanks. I tend to use the caps-wise replacement so either I didn't do it this time or it failed.

**Comment by chienandalu** [comment]
All changes done

**Review by pedrobaeza — APPROVED** [review]

**Pull request merged**